packages/console/src/components/Transfer/DraggableItem.tsx

```diff
@@ -0,0 +1,108 @@
+import type { Identifier, XYCoord } from 'dnd-core';
+import React, { ReactNode, useContext, useEffect, useRef } from 'react';
+import { useDrag, useDrop } from 'react-dnd';
+
+import { DragDropContext } from './DragDropProvider';
+
+type Props = {
+  id: string;
+  sortIndex: number;
+  moveItem: (dragIndex: number, hoverIndex: number) => void;
+  children: ReactNode;
+};
+
+type DragItemProps = {
+  sortIndex: number;
+  id: string;
+  type: string;
+};
+
+const dragType = 'TransferItem';
+
+const DraggableItem = ({ id, children, sortIndex, moveItem }: Props) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const { setIsDragging } = useContext(DragDropContext);
+  const [{ handlerId }, drop] = useDrop<DragItemProps, void, { handlerId: Identifier | undefined }>(
+    {
+      accept: dragType,
+      collect(monitor) {
+        return {
+          handlerId: monitor.getHandlerId() ?? undefined,
+        };
+      },
+      hover(item: DragItemProps, monitor) {
+        if (!ref.current) {
+          return;
+        }
+        const dragIndex = item.sortIndex;
+        const hoverIndex = sortIndex;
+
+        // Don't replace items with themselves
+        if (dragIndex === hoverIndex) {
+          return;
+        }
+
+        // Determine rectangle on screen
+        const hoverBoundingRect = ref.current.getBoundingClientRect();
+
+        // Get vertical middle
+        const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+        // Determine mouse position
+        const clientOffset = monitor.getClientOffset();
+
+        // Get pixels to the top
+        const hoverClientY = (clientOffset as XYCoord).y - hoverBoundingRect.top;
+
+        // Only perform the move when the mouse has crossed half of the items height
+        // When dragging downwards, only move when the cursor is below 50%
+        // When dragging upwards, only move when the cursor is above 50%
+
+        // Dragging downwards
+        if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+          return;
+        }
+
+        // Dragging upwards
+        if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+          return;
+        }
+
+        // Time to actually perform the action
+        moveItem(dragIndex, hoverIndex);
+
+        // Note: we're mutating the monitor item here!
+        // Generally it's better to avoid mutations,
+        // but it's good here for the sake of performance
+        // to avoid expensive index searches.
+        // eslint-disable-next-line @silverhand/fp/no-mutation
+        item.sortIndex = hoverIndex;
+      },
+    }
+  );
```

Comment:
Is this a hook config?? Why is this hook so damn hard to initialize? A lot of logics above seems should be provided out of the box, I think

Reply:
So sad but true, we need to write those code, copied from offical examples.

Reply:
Well, I'll see if I can refactor it with `dnd-kit`. The API seems way better than this